nextest-runner/src/cargo_config.rs

```diff
@@ -241,6 +241,23 @@ impl CargoConfigs {
         &self.cwd
     }
 
+    /// The environment variables to set when running Cargo commands.
+    pub fn env(&self) -> Result<BTreeMap<String, String>, CargoConfigError> {
+        let env = self
+            .discovered_configs()?
+            .filter_map(|config| match config {
+                DiscoveredConfig::CliOption { config, source: _ }
+                | DiscoveredConfig::File { config, source: _ } => Some(config),
+                DiscoveredConfig::Env => None,
+            })
+            .flat_map(|config| {
```

Comment:
Hmm, how would precedence work here? I think if you want to follow this approach to building up the environment map, you'll want to add a `.rev()` here.